morebits.js

```diff
@@ -1380,11 +1380,20 @@ Morebits.wiki.api.prototype = {
 
 		++Morebits.wiki.numberOfActionsLeft;
 
+		var queryString = $.map(q, function(val, i) {
+			if (Array.isArray(val)) {
+				return encodeURIComponent(i) + '=' + val.map(encodeURIComponent).join('|');
+			} else if (val !== undefined) {
+				return encodeURIComponent(i) + '=' + encodeURIComponent(val);
+			}
+		}).join('&').replace(/^(.*?)(token=[^&]*)&(.*)/, '$1$3&$2');
```

Comment:
Okay there's actually another issue here @siddharthvp — for queries with `intoken` (e.g. editing) this leaves the `in` behind for another parameter.  Not good.  We're not going to overhaul the token system right now, so for now I think this should do it:

```suggestion
		}).join('&').replace(/^(.*?)((?:in)?token=[^&]*)&(.*)/, '$1$3&$2');
```

But I'm not 100% it's good enough.

Reply:
Actually, I think that's technically backwards; we *don't* want to move `intoken` (well, I suppose it doesn't really matter) so it'd be better to do something like either:

```suggestion
		}).join('&').replace(/^(.*?)[&\?](token=[^&]*)&(.*)/, '$1$3&$2');
```

or

```suggestion
		}).join('&').replace(/^(.*?)([^in]token=[^&]*)&(.*)/, '$1$3&$2');
```

Reply:
How about using `\b` ? It has zero-length match, so I _think_ that should be ideal.

Reply:
Mmm, yes, I think that's probably right/best/parsimonious

Reply:
appears to work fine, pushed.

Reply:
Is there a difference between `\b(token...` and `(\btoken...`?  I'm not thinking there should be off hand.

Reply:
`\b` has a zero-length match, so no.

Reply:
Yeah - I was wondering more about ZWSP and exotics like that, I curious how they "counted."  Turns out they're `\S` and `\W!